src/main.c

```diff
@@ -111,39 +111,64 @@ void baseDefenderTask(void *argument) {
   Spaceship base_defender =
       get_spaceship_mutex(embedded_spaceships, embedded_ship->index);
   uint32_t startTime = osKernelGetTickCount(); // Temps de départ
-  int direction = 1; // 1 pour avancer, -1 pour reculer
-  int isVerticalMovementComplete =
-      0; // 0 pour mouvement vertical en cours, 1 pour terminé
+  int state = 0; // État du mouvement trapézoïdal
+  move(base_defender.ship_id, 0, 3000);
 
   while (1) {
     base_defender = update_spaceship_mutex(base_defender, embedded_spaceships,
                                            embedded_ship->index);
-    fire(base_defender.ship_id, 90);
+    Spaceship *ennemy_ships =

```

Comment:
on a qu'un seul vaisseau ici donc on enlève le pluriel
```suggestion
    Spaceship *ennemy_ship =

```